supported-configurations.json

```diff
@@ -0,0 +1,268 @@
+{
+  "__comment": "PLEASE KEEP THE KEYS SORTED :)",
+  "version": 2,
+  "supportedConfigurations": {
+    "DD_AGENT_HOST": [
+      {
+        "version": "A",
+        "type": "string",
+        "default_value": "localhost",
+        "propertyKeys": [
+          "agent_host"
+        ]
```

Comment:
I don't think either `type` or `propertyKeys` are going to be used to generate code (or I don't see how we would do it?). Is it a good idea to define it here?